token-lending/program/src/state/reserve.rs

```diff
@@ -440,6 +450,24 @@ impl ReserveLiquidity {
         }
         self.borrowed_amount_wads.try_div(total_supply)?.try_into()
     }
+
+    /// Record the amount that is in the flash loan.
+    pub fn flash_loan_start(&mut self, borrow_amount: u64) -> ProgramResult {
+        if borrow_amount > self.available_amount {
+            return Err(LendingError::InsufficientLiquidity.into());
+        }
+
+        self.available_amount -= borrow_amount;
```

Comment:
Whenever this value changes, `total_supply` will be adjusted. `total_supply` should be adjusted to account for active flash loans. It's worth thinking about what kind of damage someone with a flash loan could do to the lending protocol itself

Reply:
I see, I missed this details. I have now reduce the amount of collateral correspondingly to avoid the change in exchange rate during flash loan.

Reply:
There is another side to total supply that should be considered. It's also used when determining the borrow rate so we need to ensure that the adjusted borrow rate can't be taken advantage of

Reply:
> It's also used when determining the borrow rate so we need to ensure that the adjusted borrow rate can't be taken advantage of

@leonardgee let's take another close look at how this works in the current implementation